### DISCUSSION TITLE
Extend the Billing and Shipping Addresses Schema on the Store API

### DISCUSSION THREAD

**Discussion by gustavoconter** [woocommerce/woocommerce#45753] [WooCommerce, Blocks, and Site Editing]
I am part of a team working on a headless WooCommerce store, specifically focusing on developing a new checkout page. Our store operates in Brazil, and traditionally, we have included additional fields in our checkout process, such as billing_number and billing_neighborhood, to accommodate local address formats better.

While integrating the Store API for managing carts, we've encountered a need to extend the address schema to include these extra fields. Here's an example of our current schema for reference:

```json
"billing_address": {
  "first_name": "",
  "last_name": "",
  "company": "",
  "address_1": "",
  "address_2": "",
  "city": "",
  "state": "SP",
  "postcode": "",
  "country": "BR",
  "email": "",
  "phone": ""
}
```

Our goal is to enhance this schema to incorporate additional fields, specifically aiming for a target schema like this:

```json
"billing_address": {
  "first_name": "",
  "last_name": "",
  "company": "",
  "address_1": "",
  "address_2": "",
  "city": "",
  "state": "SP",
  "postcode": "",
  "country": "BR",
  "email": "",
  "phone": "",
  "neighborhood": ""
}
```

Can we extend the address schema within the Store API to include custom fields like billing_neighborhood?
Also, is it possible to make these custom fields savable using the `/update-customer` endpoint?

**Comment by opr** [comment]
Hi @gustavoconter - we're working on a feature that will allow you to register custom fields, [here's the relevant discussion](https://github.com/woocommerce/woocommerce/discussions/42995).

~~We expect this to be released in WooCommerce 8.8 but it will still be in an experimental phase at that point.~~

Edit: This is out in 8.7 which released yesterday. It is still in experimental phase though so be aware things _could_ change. [Here's some documentation for it.](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md)

Please follow the discussion I linked above for more information and please keep an eye on [our dev blog](https://developer.woo.com/blog/) for information relating to this feature as we will publish something when we release it.

**Reply by gustavoconter** [reply]
Hi @opr, thanks for the quick reply!

Great to know that this feature is already being developed, will keep an eye on the blog as well!
